### PR TITLE
Add support for YDKB Chili

### PR DESCRIPTION
**Pull request by TerryMathews**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/chili/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+[Chili PCB](https://gd3.alicdn.com/imgextra/i3/12420845/O1CN01ilPFVn1I73GjalIXf_!!12420845.jpg)
+
+
+The YDKB Chili is a Cherry G80-3000 replacement PCB utilizing the ATmega32U4 microcontroller.
+
+Keyboard Maintainer: QMK community  
+Hardware Supported: YDKB Chili
+Hardware Availability: [TaoBao](https://item.taobao.com/item.htm?id=565823984744)  
+
```
Comment (fauxpark) [inline keyboards/chili/README.md:11]:
```suggestion
The YDKB Chili is a Cherry G80-3000 replacement PCB utilizing the ATmega32U4 microcontroller.

* Keyboard Maintainer: QMK community
* Hardware Supported: YDKB Chili
* Hardware Availability: [TaoBao](https://item.taobao.com/item.htm?id=565823984744)
```

File: keyboards/chili/chili.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "chili.h"
+#include "led.h"
```
Comment (fauxpark) [inline keyboards/chili/chili.c:17]:
```suggestion
```

File: keyboards/chili/chili.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    // put your per-action keyboard code here
+    // runs for every action, just before processing by the firmware
+
+    return process_record_user(keycode, record);
+}
+
```
Comment (fauxpark) [inline keyboards/chili/chili.c:39]:
```suggestion
```

File: keyboards/chili/chili.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+void led_init_ports(void) {
+    setPinOutput(B1);
+    writePinHigh(B1);
+	setPinOutput(B2);
+    writePinHigh(B2);
+	setPinOutput(B3);
```
Comment (fauxpark) [inline keyboards/chili/chili.c:45]:
```suggestion
    setPinOutput(B2);
    writePinHigh(B2);
    setPinOutput(B3);
```

File: keyboards/chili/chili.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    } else {
+        writePinHigh(B2);
+    }
+	if (usb_led & (1 << USB_LED_SCROLL_LOCK)) {
+        writePinLow(B3);
+    } else {
+        writePinHigh(B3);
+    }
+    led_set_user(usb_led);
+}
```
Comment (fauxpark) [inline keyboards/chili/chili.c:66]:
```suggestion
bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)) {
        writePin(B1, !led_state.num_lock);
        writePin(B2, !led_state.caps_lock);
        writePin(B3, !led_state.scroll_lock);
    }
}
```

File: keyboards/chili/chili.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_ALL( \
```
Comment (fauxpark) [inline keyboards/chili/chili.h:20]:
```suggestion
#define LAYOUT_all( \
```

File: keyboards/chili/chili.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09 }, \
+    { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19 }, \
+    { K20, K21, K22, K23, K24, K25, K26, K27, K28, K29 }, \
+    { K30, K31, K32, K33, K34, K35, K36, K37, K38, K39 }, \
+    { K40, K41, K42, K43, K44, K45, K46, K47, K48, K49 }, \
+    { K50, K51, K52, K53, K54, K55, K56, K57, K58, K59 }, \
+	{ K60, K61, K62, K63, K64, K65, K66, K67, K68, K69 }, \
+	{ K70, K71, K72, K73, K74, K75, K76, K77, K78, K79 }, \
+	{ K80, K81, K82, K83, K84, K85, K86, K87, K88, K89 }, \
+	{ K90, K91, K92, K93, K94, K95, K96, K97, K98, K99 }, \
```
Comment (fauxpark) [inline keyboards/chili/chili.h:37]:
```suggestion
    { K60, K61, K62, K63, K64, K65, K66, K67, K68, K69 }, \
    { K70, K71, K72, K73, K74, K75, K76, K77, K78, K79 }, \
    { K80, K81, K82, K83, K84, K85, K86, K87, K88, K89 }, \
    { K90, K91, K92, K93, K94, K95, K96, K97, K98, K99 }, \
```

File: keyboards/chili/info.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "YDKB Chili", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "width": 22.5, 
+    "height": 6.5, 
+    "layouts": {
+        "LAYOUT_ALL": {
+			"key_count": 110,
```
Comment (fauxpark) [inline keyboards/chili/info.json:9]:
```suggestion
```

File: keyboards/chili/info.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "YDKB Chili", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "width": 22.5, 
+    "height": 6.5, 
+    "layouts": {
+        "LAYOUT_ALL": {
```
Comment (fauxpark) [inline keyboards/chili/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/chili/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______, \
+    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,   \
+    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______, \
+    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______           \
```
Comment (fauxpark) [inline keyboards/chili/keymaps/default/keymap.c:41]:
```suggestion
  [BASE] = LAYOUT_all(
    KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,       KC_F5,   KC_F6,   KC_F7,   KC_F8,        KC_F9,   KC_F10,  KC_F11,  KC_F12,             KC_PSCR, KC_SLCK, KC_PAUS,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_NO,              KC_INS,  KC_HOME, KC_PGUP,     KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
    KC_TAB,    KC_Q,     KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,     KC_LBRC, KC_RBRC,      KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,     KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
    KC_CAPS,      KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,              KC_ENT,                                            KC_P4,   KC_P5,   KC_P6,   KC_NO,
    KC_LSFT, KC_NO,  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,             KC_NO, KC_RSFT,                     KC_UP,                KC_P1,   KC_P2,   KC_P3,   KC_PENT,
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_NO, KC_SPC,                               KC_RALT, KC_RGUI, KC_APP,  KC_RCTL,            KC_LEFT, KC_DOWN, KC_RGHT,     KC_P0,   KC_NO,   KC_PDOT
  ),
  [FUNCTION] = LAYOUT_all(
    _______,           _______,   _______,   _______,   _______,       _______,   _______,   _______,   _______,        _______,   _______,  _______,  _______,             _______, _______, _______,
    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______, _______,              _______,  _______, _______,     _______, _______, _______, _______,
    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______,
    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,
    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______,
    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______
```

File: keyboards/chili/rules.mk [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+
```
Comment (fauxpark) [inline keyboards/chili/rules.mk:17]:
```suggestion
```

File: keyboards/chili/rules.mk [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight. Do not enable this with audio at the same time.
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+OLED_DRIVER_ENABLE = no
```
Comment (fauxpark) [inline keyboards/chili/rules.mk:33]:
```suggestion
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight
MIDI_ENABLE = no            # MIDI controls
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/chili/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______, \
+    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,   \
+    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______, \
+    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______           \
```
Comment (fauxpark) [inline keyboards/chili/keymaps/via/keymap.c:59]:
```suggestion
  [BASE] = LAYOUT_all(
    KC_ESC,           KC_F1,   KC_F2,   KC_F3,   KC_F4,       KC_F5,   KC_F6,   KC_F7,   KC_F8,        KC_F9,   KC_F10,  KC_F11,  KC_F12,             KC_PSCR, KC_SLCK, KC_PAUS,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_NO,              KC_INS,  KC_HOME, KC_PGUP,     KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
    KC_TAB,    KC_Q,     KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,     KC_LBRC, KC_RBRC,      KC_BSLS,            KC_DEL,  KC_END,  KC_PGDN,     KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
    KC_CAPS,      KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,              KC_ENT,                                            KC_P4,   KC_P5,   KC_P6,   KC_NO,
    KC_LSFT, KC_NO,  KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,             KC_NO, KC_RSFT,                     KC_UP,                KC_P1,   KC_P2,   KC_P3,   KC_PENT,
    KC_LCTL, KC_LGUI, KC_LALT,                            KC_NO, KC_SPC,                               KC_RALT, KC_RGUI, KC_APP,  KC_RCTL,            KC_LEFT, KC_DOWN, KC_RGHT,     KC_P0,   KC_NO,   KC_PDOT
  ),
  [FUNCTION] = LAYOUT_all(
    _______,           _______,   _______,   _______,   _______,       _______,   _______,   _______,   _______,        _______,   _______,  _______,  _______,             _______, _______, _______,
    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______, _______,              _______,  _______, _______,     _______, _______, _______, _______,
    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______,
    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,
    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______,
    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______
  ),
  [ALTERNATE] = LAYOUT_all(
    _______,           _______,   _______,   _______,   _______,       _______,   _______,   _______,   _______,        _______,   _______,  _______,  _______,             _______, _______, _______,
    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______, _______,              _______,  _______, _______,     _______, _______, _______, _______,
    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______,
    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,
    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______,
    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______
  ),
  [LAST] = LAYOUT_all(
    _______,           _______,   _______,   _______,   _______,       _______,   _______,   _______,   _______,        _______,   _______,  _______,  _______,             _______, _______, _______,
    _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______, _______,              _______,  _______, _______,     _______, _______, _______, _______,
    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______,
    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,
    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______,
    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______
```

File: keyboards/chili/chili.c (hunk 5 above)
Comment (TerryMathews) [reply keyboards/chili/chili.c]: @fauxpark I can't apply these changes for some reason. Can you re-suggest them so I can apply it?

File: keyboards/chili/README.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
+YDKB Chili
```
Comment (zvecr) [inline keyboards/chili/README.md:1]:
```suggestion
# YDKB Chili
```

File: keyboards/chili/chili.c [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
```
Comment (zvecr) [inline keyboards/chili/chili.c:51]:
```suggestion
bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if(res) {
        writePin(B1, !led_state.num_lock);
        writePin(B2, !led_state.caps_lock);
        writePin(B3, !led_state.scroll_lock);
    }
    return res;
}
```

File: keyboards/chili/chili.c (hunk 5 above)
Comment (zvecr) [reply keyboards/chili/chili.c]: done

File: keyboards/chili/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    _______,    _______,     _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,     _______, _______,      _______,            _______,  _______,  _______,     _______,   _______,   _______,   _______,
+    _______,      _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,              _______,                                            _______,   _______,   _______,   _______,
+    _______, _______,  _______,    _______,    _______,    _______,    _______,    _______,    _______,    _______, _______,  _______,             _______, _______,                     _______,                _______,   _______,   _______,   _______,
+    _______, _______, _______,                            _______, _______,                               _______, _______, _______,  _______,            _______, _______, _______,     _______,   _______,   _______
+  ),
```
Comment (Erovia) [inline keyboards/chili/keymaps/default/keymap.c:42]:
```suggestion
 
```
An empty layer is not very useful, especially if there is no way to activate it.

File: keyboards/chili/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+ */
+ 
+#include QMK_KEYBOARD_H
+
+//Layers
+
+enum {
+  BASE = 0,
+  FUNCTION,
+};
```
Comment (Erovia) [inline keyboards/chili/keymaps/default/keymap.c:24]:
```suggestion
  BASE,
};
```

File: keyboards/chili/config.h [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define QMK_ESC_OUTPUT D4 // usually COL
+#define QMK_ESC_INPUT F5 // usually ROW
+#define QMK_LED B1 // NumLock on TKC1800
+//#define QMK_SPEAKER C6
```
Comment (Erovia) [inline keyboards/chili/config.h:73]:
```suggestion

```
These only work with qmk-dfu bootloader.

**Comment by drashna** [comment]
@TerryMathews it looks like you missed a few suggestions, that were hidden.  These need to be applied to fix the Travis CI compiler issues.

**Comment by TerryMathews** [comment]
Sorry about that, found them and applied them.

File: keyboards/chili/README.md [hunk 19]
```diff
@@ -0,0 +1,15 @@
+# YDKB Chili
+
+[Chili PCB](https://gd3.alicdn.com/imgextra/i3/12420845/O1CN01ilPFVn1I73GjalIXf_!!12420845.jpg)
```
Comment (fauxpark) [inline keyboards/chili/README.md:3]: This image doesn't seem to be loading in the Markdown preview, could you rehost it on imgur?

File: keyboards/chili/README.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
+# YDKB Chili
+
+[Chili PCB](https://imgur.com/fKi896a)
```
Comment (fauxpark) [inline keyboards/chili/README.md:3]:
```suggestion
[Chili PCB](https://i.imgur.com/fKi896a.jpg)
```
Needs to be a direct link.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**